Dockerfile

```diff
@@ -2,14 +2,17 @@ FROM python:3.8-slim-buster as base
 
 WORKDIR /app
 COPY poetry.lock pyproject.toml /app/
-RUN apt-get update && apt-get install -y curl && pip install poetry && poetry install --no-root 
+RUN apt-get update && apt-get install -y curl && pip install poetry 
+RUN poetry config virtualenvs.create false --local && poetry install --no-root
 EXPOSE 5000
 
 FROM base as production
 COPY ./todo_app /app/todo_app
 ENV PYTHONPATH=/app
 RUN poetry add gunicorn
-ENTRYPOINT ["poetry","run","gunicorn","-w","4","-b","0.0.0.0","todo_app.app:app"]
+ENV PORT=8000
+EXPOSE $PORT
+ENTRYPOINT "poetry run gunicorn -w 4 -b 0.0.0.0:$PORT todo_app.app:app"
```

Comment:
Note that this change to the shell format breaks running the container locally - see the project exercise hints for moving to an entrypoint script to get more consistent behaviour for your container between local and heroku

Reply:
As advised moved to entrypoint script